nrf-hal-common/src/uarte.rs

```diff
@@ -43,6 +43,16 @@ where
     T: Instance,
 {
     pub fn new(uarte: T, mut pins: Pins, parity: Parity, baudrate: Baudrate) -> Self {
+        // Is the UART already on? It might be if you had a bootloader
+        if uarte.enable.read().bits() != 0 {
+            while uarte.events_txstopped.read().bits() == 0 {
```

Comment:
`events_txstopped` only fires if whoever was using it before did a `stoptx`. This might spin forever if that's not the case.

Reply:
This is exactly why the errata workaround checks for rxenable/txenable, but we can't rely on those being present in nrf52 since they're undocuemnted...

Reply:
Ah, I see. With no check, I splatter the last line emitted by SPM. What's a good thing to check for here? Or do we not care that we configure the UART whilst it may still be enabled?

Reply:
there might be side effects if it's not disabled yeah :S 

Maybe always do a stoptx? I think if tx is stopped stoptx still causes a new txstopped, so the loop is guaranteed to hang? I'm not 100% sure though

Reply:
OK, will do.